dotcom-rendering/src/components/Discussion/Comments.tsx

```diff
@@ -257,6 +270,10 @@ export const Comments = ({
 		} else {
 			handleFilterChange(newFilterObject);
 		}
+
+		isWeb &&
+			enableMobileDiscussionAdsSwitch &&
+			document.dispatchEvent(new CustomEvent('comments-state-change'));
```

Comment:
You could wrap this in its own function if you do not want to literal strings peppered in different places, for example:

```ts
/** @see https://github.com/guardian/commercial/blob/9175e572a5a81e334208775c64b7b479dd544c6c/src/insert/comments-expanded-advert.ts#L197 */
const dispatchCommentsStateChange = () =>
  document.dispatchEvent(new CustomEvent('comments-state-change'));
  
/* … later … */

isWeb && enableMobileDiscussionAdsSwitch &&
  dispatchCommentsStateChange();

```